src/Components/Locations.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react'
+import axios from 'axios'
+
+export class Locations extends Component {
+constructor() {
+    super();
+    this.state = {
+        locations: [],
+        showList: false
+    }
+}
+
+getLocations = async () => {
+    const { data } = await axios.get(`https://ghibliapi.herokuapp.com/locations`);
+
+    this.setState({
+        locations: data.map((loc )=> loc)
+    })
+}
+handleClick = () => {
+    this.setState({
+      showList: !this.state.showList,
+    });
+  };
+
+  componentDidMount() {
+    this.getLocations();
+  }
+
+  render() {
+    const { locations, showList } = this.state;
+    const listItems = locations.map((loc, i) => <li key={i}>{
+        loc.name} <br/>
+        {loc.terrain}<br/>
+        {loc.climate} </li>);
+
+        return (
+            <div  style={{
+                backgroundColor: 'grey',
+                width: 'auto',
+                height: '500px'
+              }}
+            >
+                <h1>List of Locations</h1>
+                <button onClick={this.handleClick}>
+          {showList ? "Hide Locations" : "Show Locations"}
+        </button>
+
+        <ul>{showList ? listItems : null}</ul>
```

Comment:
On this line, there seems to be an issue with code indent alignment. It's very important for each element to be indented to the correct line as to make the code much more readable. This is something that is that is closely looked at for technical interviews and code submissions. I also noticed missing semi colons. While it seems optional most of the time, there are occasions where it can cause a huge headache because the code got evaluated incorrectly because of a missing semicolon.